course_catalog/filters.py

```diff
@@ -39,3 +42,30 @@ def filter_audience(self, queryset, _, value):
         else:
             queryset = queryset.exclude(platform__in=PROFESSIONAL_COURSE_PLATFORMS)
         return queryset
+
+    def filter_withcertificate(self, queryset, _, value):
+        """Certificate filter for courses"""
+        if value == "":
+            return queryset
+        else:
+            qs1 = queryset.filter(
+                runs__availability__in=[
+                    AvailabilityType.current.value,
+                    AvailabilityType.upcoming.value,
+                    AvailabilityType.starting_soon.value,
+                ],
+                platform=PlatformType.mitx.value,
+            )
+            qs2 = queryset.filter(
+                runs__availability__in=[
+                    AvailabilityType.current.value,
+                    AvailabilityType.upcoming.value,
+                    AvailabilityType.starting_soon.value,
+                ],
+                platform__in=PROFESSIONAL_COURSE_PLATFORMS,
```

Comment:
If we are making the proposed change to show only professional courses with non-archived availability, you don't need q1 and q2 you can just have one query with  `platform__in=PROFESSIONAL_COURSE_PLATFORMS.append(PlatformType.mitx.value)`

Reply:
yeah, unintentional, BUT I was able to drop it into a Q or I think and simplify it a ton